src/screens/Exchange/Exchange.js

```diff
@@ -242,106 +224,61 @@ class ExchangeScreen extends React.Component<Props, State> {
         getInputRef={ref => { this.fromInputRef = ref; }}
-        onChange={this.handleFromInputChange}
-        value={value}
-        onFocus={this.onFocusInput}
         onBlur={this.blurFromInput}
-        errorMessage={errorMessage}
-        asset={fromAsset}
-        onAssetPress={() => this.setState({ showSellOptions: true })}
-        labelText={assetBalance && getFormattedSellMax(fromAsset)}
-        onLabelPress={() => this.handleUsePercent(100)}
-        leftSideText={displayFiatFromAmount
-          ? t('tokenValue', {
-            value: formatAmount(fromAmount || '0', 2) || '0',
-            token: fromAsset.symbol || '',
-          })
-          : formatFiat(fromAmountInFiat, baseFiatCurrency).replace(/ /g, '')
-        }
-        leftSideSymbol="-"
-        onLeftSideTextPress={() => this.setState({ displayFiatFromAmount: !displayFiatFromAmount })}
-        rightPlaceholder={displayFiatFromAmount ? baseFiatCurrency || defaultFiatCurrency : symbol}
-        inputAccessoryViewID={INPUT_ACCESSORY_NATIVE_ID}
+        onFormValid={valid => this.setState({ isFormValid: valid })}
       />
     );
   };
 
   getToInput = () => {
-    const { baseFiatCurrency, offers, rates } = this.props;
+    const { offers } = this.props;
     const {
-      displayFiatToAmount, toAsset, fromAmount,
+      toAsset, fromAmount,
     } = this.state;
+    const { toOptions, horizontalOptions } = this.options;
 
-    let value;
-    let toAmount;
-    let toAmountInFiat;
+    let toAmount = '0';
     if (offers?.length && fromAmount) {
-      toAmount = getBestAmountToBuy(offers, fromAmount);
-      if (toAmount) {
-        toAmountInFiat = formatAmount(getBalanceInFiat(baseFiatCurrency, toAmount, rates, toAsset.symbol || ''), 2);
-        value = displayFiatToAmount ? toAmountInFiat : formatAmount(toAmount || '', 6);
-      }
+      toAmount = getBestAmountToBuy(offers, fromAmount) || '0';
     }
-    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
 
     return (
-      <TextInput
+      <ValueInput
         disabled
-        value={value}
-        onBlur={this.blurFromInput}
-        asset={toAsset}
-        onAssetPress={() => this.setState({ showBuyOptions: true })}
-        leftSideText={displayFiatToAmount
-          ? t('tokenValue', { value: formatAmount(toAmount || '0', 2), token: toAsset.symbol || '' })
-          : formatFiat(toAmountInFiat || '0', fiatCurrency).replace(/ /g, '')
-        }
+        value={toAmount}
+        assetData={toAsset}
+        onAssetDataChange={(assetData) => this.setState({ toAsset: assetData })}
+        selectorOptionsTitle={t('label.buy')}
+        customAssets={toOptions}
+        horizontalOptions={horizontalOptions}
         leftSideSymbol="+"
-        onLeftSideTextPress={() => this.setState({ displayFiatToAmount: !displayFiatToAmount })}
-        rightPlaceholder={displayFiatToAmount ? fiatCurrency : toAsset.symbol || ''}
+        onBlur={this.blurFromInput}
+        hideMaxSend
```

Comment:
Based on designs, it shouldn't be hidden. 
Although, I'd argue that it isn't needed here, too.
At least not clickable :)

Reply:
https://zpl.io/awKZWGd